src/client/views/HotspotVote.js

```diff
@@ -0,0 +1,95 @@
+import React, { Fragment, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import View from '~/client/components/View';
+import VoteSession from '~/client/components/VoteSession';
+import notify, {
+  NotificationsTypes,
+} from '~/client/store/notifications/actions';
+import translate from '~/common/services/i18n';
+import { initializeVote, resetVote } from '~/client/store/vote/actions';
+import { useResource } from '~/client/hooks/requests';
+import Spinner from '~/client/components/Spinner';
+import { HeadingPrimaryStyle } from '~/client/styles/typography';
+import { ContainerStyle } from '~/client/styles/layout';
+import ColorSection from '~/client/components/ColorSection';
+import Header from '~/client/components/Header';
+import { isAvailable, hasItem, setItem } from '~/client/utils/storage';
+
+const HotspotVote = () => {
+  const dispatch = useDispatch();
+  const vote = useSelector((state) => state.vote);
+  const [isError, setIsError] = useState(false);
+
+  const isVoteReady = !!vote.address;
+
+  const [voteData, isVoteDataLoading] = useResource(['hotspot'], {
+    onError: () => {
+      setIsError(true);
+      dispatch(
+        notify({
+          text: translate('HotspotVote.errorNotEnabled'),
+        }),
+      );
+    },
+  });
+
+  useEffect(() => {
+    if (isVoteDataLoading || isError) {
+      return;
+    }
+
+    if (isAvailable && hasItem(`hasVoted-${voteData.festivalQuestionId}`)) {
+      setIsError(true);
+      dispatch(
+        notify({
+          text: translate('HotspotVote.errorAlreadyVoted'),
+          type: NotificationsTypes.ERROR,
+        }),
+      );
+    }
+
+    try {
+      dispatch(resetVote());
+      dispatch(initializeVote(voteData));
+      setItem(`hasVoted-${voteData.festivalQuestionId}`, true);
+    } catch (error) {
+      dispatch(
+        notify({
+          text: translate('HotspotVote.errorInvalidVoteData'),
+          type: NotificationsTypes.ERROR,
+        }),
+      );
+    }
```

Comment:
So I'm just double checking if it's safe to run the `try .. catch` clause, even if an error was detected in the branch before. I'm not sure the implications of `resetVote` and `initializeVote` and whether they should run even if the vote has been already cast.

Reply:
So this should actually be safe to run (it doesn't vote, just clears and recreates the vote private key in localstorage, which is disposable) - but glad you highlighted it because looking through it I think I was setting the localstorage token in the wrong space (it shouldn't happen when the vote gets initialized, but actually when it is cast). Also added a return after dispatching the error when `hasVoted` is true, just to be safe